src/graph.rs

```diff
@@ -348,215 +428,152 @@ impl Dependency {
   /// Optionally return the module specifier in the module graph that points to
   /// the "code" dependency in the graph.
   pub fn get_code(&self) -> Option<&ModuleSpecifier> {
-    match &self.maybe_code {
-      Some(Ok((specifier, _))) => Some(specifier),
-      _ => None,
-    }
+    self.maybe_code.maybe_specifier()
   }
 
   /// Optionally return the module specifier in the module graph that points to
   /// the type only dependency in the graph.
   pub fn get_type(&self) -> Option<&ModuleSpecifier> {
-    match &self.maybe_type {
-      Some(Ok((specifier, _))) => Some(specifier),
-      _ => None,
-    }
+    self.maybe_type.maybe_specifier()
   }
 
   /// Check to see if the position falls within the range of the code or types
   /// entry for the dependency, returning a reference to the range if true,
   /// otherwise none.
   pub fn includes(&self, position: &Position) -> Option<&Range> {
-    match &self.maybe_code {
-      Some(Ok((_, range))) if range.includes(position) => return Some(range),
-      Some(Err(err)) if err.range().includes(position) => {
-        return Some(err.range())
-      }
-      _ => (),
-    }
-    match &self.maybe_type {
-      Some(Ok((_, range))) if range.includes(position) => Some(range),
-      Some(Err(err)) if err.range().includes(position) => Some(err.range()),
-      _ => None,
+    let code = self.maybe_code.includes(position);
+    if code.is_none() {
+      self.maybe_type.includes(position)
+    } else {
+      code
     }
   }
 }
 
-#[derive(Debug, Clone, Serialize)]
-#[serde(untagged)]
-pub enum Module {
-  Es(Box<EsModule>),
-  Synthetic(Box<SyntheticModule>),
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum ModuleKind {
+  /// An asserted module, associated with the specific type (e.g. "JSON").
+  /// Dependency analysis does not occur on asserted modules.
+  Asserted(String),
+  /// An AMD module. Currently dependency analysis is not supported for these
+  /// kinds of modules.
+  AMD,
+  /// A CommonJS module.
+  CommonJs,
+  /// An ECMAScript Module (JavaScript Module).
+  Esm,
+  /// A JavaScript script module. A slight misnomer, but it allows "plain"
+  /// scripts to be imported into the module graph, but without supporting any
+  /// dependency analysis.
+  Script,
+  /// An injected module where the any dependencies are asserted and no
```

Comment:
```suggestion
  /// An injected module where no dependencies are asserted and no
```

Reply:
No, there are specifically dependencies, the whole purpose of a synthetic module.  It should be "where any dependencies are asserted".